rasa/core/policies/intent_ted_policy.py

```diff
@@ -0,0 +1,688 @@
+
+        return labels_embed
+
+    def compute_thresholds(
+        self, model_data: RasaModelData, label_ids: np.ndarray
+    ) -> Dict[int, float]:
+        """Compute prediction thresholds for each intent.
+
+        These thresholds are used at inference time to predict
+        whether a query intent is likely or not.
+
+        Args:
+            model_data: Data used during model training.
+            label_ids: Numerical IDs of labels corresponding to data points used during training.
+
+        Returns:
+            Computed thresholds for each intent label present in `label_ids`.
+        """
+        self._training = False
+
+        batch_size = (
+            self.config[BATCH_SIZES]
+            if isinstance(self.config[BATCH_SIZES], int)
+            else self.config[BATCH_SIZES][0]
+        )
+        outputs = self.run_inference(model_data, batch_size=batch_size)
+
+        # Collect scores across all data points
+        label_id_scores = self._collect_label_id_similarities_from_outputs(
+            label_ids, outputs
+        )
+
+        return self._pick_threshold_from_similarities(label_id_scores)
+
+    @staticmethod
+    def _pick_threshold_from_similarities(
+        label_id_similarities: Dict[int, List[float]]
+    ) -> Dict[int, float]:
+        """Compute threshold for predicted similarities.
+
+        The threshold for an intent is computed as the minimum
+        of all similarities predicted for that particular intent.
+
+        Args:
+            label_id_similarities: Similarities predicted for each label/
+
+        Returns:
+            Computed thresholds for each intent label.
+        """
+        return {
+            label_id: min(label_id_similarities[label_id])
+            for label_id in label_id_similarities
+        }
+
+    @staticmethod
+    def _collect_label_id_similarities_from_outputs(
+        label_ids: np.ndarray, outputs: Dict[Text, Union[np.ndarray, Dict[Text, Any]]]
+    ) -> Dict[int, List[float]]:
+        """Collect similarities predicted for each label id.
+
+        Args:
+            label_ids: Numerical IDs of labels for each data point used during training.
+            outputs: Model's predictions for each data point.
+
+        Returns:
+            Similarities grouped by intent label id.
+        """
+        label_id_scores = {}
+        for index, all_pos_labels in enumerate(label_ids):
+
+            # Take the contribution of only first label id
+            # because `outputs` will contain another
+            # data point where remaining label ids will
+            # be the first label id
+            first_pos_label_id = all_pos_labels[0]
+
+            if first_pos_label_id not in label_id_scores:
+                label_id_scores[first_pos_label_id] = []
+
+            label_id_scores[first_pos_label_id].append(
+                outputs["similarities"][index, 0, first_pos_label_id]
+            )
+        return label_id_scores
```

Comment:
Good job! This one is actually clear and straightforward, especially after going through the `TEDPolicy` :)